### PR TITLE
New global activities [SCI-3138]

### PR DESCRIPTION
**Pull request by urbanrotnik**
Jira ticket: [SCI-3138](https://biosistemika.atlassian.net/browse/SCI-3138)

### What was done
Implemented creations for new activities from this [list](https://docs.google.com/spreadsheets/d/182zm7Q1it1U2ggQ9T28r3SKTnoFQFKiiE4hc9Io94Zc/edit#gid=989483893).

#### ToDo:
Look at the [list](https://docs.google.com/spreadsheets/d/182zm7Q1it1U2ggQ9T28r3SKTnoFQFKiiE4hc9Io94Zc/edit#gid=989483893) 
- turqiz: Needs to be implemented in Add-on
- red: This scenario does not exist. 
- green: Needs dev/product grooming or just a confirmation "its good enought"
- orange: Needs to be implemented in Core 

### Note:
**This is not a working solution until all old activities won't be commend out or implemented to create in a new way!** 
@mlorb you can rebase your feature branch on this

### CODE REVIEW

File: spec/controllers/repository_rows_controller_spec.rb [hunk 1]
```diff
@@ -34,7 +36,7 @@
 
   describe '#show' do
     it 'unsuccessful response with non existing id' do
-      get :show, format: :json, params: { id: 999999 }
+      get :show, format: :json, params: { id: -1 }
```
Comment (radiokills) [inline spec/controllers/repository_rows_controller_spec.rb]: :+1:

File: app/controllers/my_modules_controller.rb [hunk 2]
```diff
@@ -208,6 +209,29 @@ def update
             module: @my_module.name
           )
         )
+      elsif saved && due_date_changes
```
[the remark this answers is not in the document]
Comment (urbanrotnik) [reply app/controllers/my_modules_controller.rb]: When there are no changes, `due_date_changes` is `nil` not `[]`

File: app/controllers/my_modules_controller.rb [hunk 3]
```diff
@@ -208,6 +209,29 @@ def update
...
+                                                 @my_module.experiment.project,
+                                               team: current_team,
+                                               message_items: {
+                                                 my_module: @my_module.id,
+                                                 my_module_duedate:
+                                                   {
+                                                     id: @my_module.id,
+                                                     value_for: 'due_date'
+                                                   }
+                                               })
```
[the remark this answers is not in the document]
Comment (urbanrotnik) [reply app/controllers/my_modules_controller.rb]: Hm.. not sure but I would refactor all activity creations in this `update` method. Will do later today

File: app/controllers/protocols_controller.rb [hunk 4]
```diff
@@ -595,6 +608,15 @@ def import
         end
       else
         format.json do
+          Activities::CreateActivityService
```
[the remark this answers is not in the document]
Comment (urbanrotnik) [reply app/controllers/protocols_controller.rb]: 👍

File: app/controllers/protocols_controller.rb [hunk 5]
```diff
@@ -1189,4 +1223,15 @@ def metadata_params
...
+  def log_activity(type_of)
+    Activities::CreateActivityService
+      .call(activity_type: type_of,
+            owner: current_user,
+            subject: @protocol,
+            team: current_team,
+            message_items: {
+              protocol: @protocol.id
+            })
+  end
```
[the remark this answers is not in the document]
Comment (urbanrotnik) [reply app/controllers/protocols_controller.rb]: You mean validating of correct `message_items` or what?

File: app/controllers/repository_columns_controller.rb [hunk 6]
```diff
@@ -273,4 +279,16 @@ def update_repository_list_items(item_names)
...
+      .call(activity_type: type_of,
+            owner: current_user,
+            subject: @repository,
+            team: current_team,
+            message_items: {
+              repository_column: @repository_column.id,
+              repository: @repository.id
+            })
+  end
 end
```
[the remark this answers is not in the document]
Comment (urbanrotnik) [reply app/controllers/repository_columns_controller.rb]: Its actual action specific...

File: app/models/experiment.rb [hunk 7]
```diff
@@ -143,7 +143,7 @@ def update_canvas(
         cloned_modules = cloned_pairs.collect { |mn, _| mn }
 
         # Rename modules
-        rename_modules(to_rename)
+        rename_modules(to_rename, current_user)\
```
[the remark this answers is not in the document]
Comment (urbanrotnik) [reply app/models/experiment.rb]: Woop, typo

File: app/controllers/experiments_controller.rb [hunk 8]
```diff
@@ -107,17 +107,22 @@ def update
...
-        experiment: @experiment,
-        user: current_user,
-        message: I18n.t(
-          'activities.edit_experiment',
-          user: current_user.full_name,
-          experiment: @experiment.name
-        )
-      )
+
+      activity_type_of = if experiment_params[:archived] == 'false'
```
[the remark this answers is not in the document]
Comment (urbanrotnik) [reply app/controllers/experiments_controller.rb]: `activity_type_of ` ?

**Comment by urbanrotnik** [comment]
Thanks @radiokills  for a careful review :) 

I fixed some things or add comments. Would be great if Anton and Alex can also check changes.

File: app/controllers/my_modules_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (radiokills) [reply app/controllers/my_modules_controller.rb]: :+1:

File: app/controllers/my_modules_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (radiokills) [reply app/controllers/my_modules_controller.rb]: :+1:

File: app/controllers/protocols_controller.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (radiokills) [reply app/controllers/protocols_controller.rb]: No, the whole method.

File: app/controllers/experiments_controller.rb (hunk 8 above)
[the remark this answers is not in the document]
Comment (radiokills) [reply app/controllers/experiments_controller.rb]: yes :)

File: app/controllers/protocols_controller.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (urbanrotnik) [reply app/controllers/protocols_controller.rb]: I am not sure how.. but yeah :) 👍

**Pull request merged**